CSC8503CoreClasses/AnimationSystem.cpp

```diff
@@ -24,87 +27,160 @@ void AnimationSystem::Clear()
 {
-	
-	UpdateAllAnimationObjects(dt);
-	UpdateMaterials();
+	UpdateCurrentFrames(dt);
+	UpdateAnimations(preAnimationList);
+	UpdateAllAnimationObjects(dt, UpdatableObjects);
+	/*UpdateMaterials();*/
 	
 }
 
-void AnimationSystem::UpdateAllAnimationObjects(float dt)
+void AnimationSystem::UpdateAllAnimationObjects(float dt, vector<GameObject*> UpdatableObjects)
 {
 	mAnimationList.clear();
-	
-	gameWorld.OperateOnContents(
-		[&](GameObject* o) {
+	mGuardList.clear();
+	mPlayerList.clear();
+		for (auto& o : UpdatableObjects) {
+			//Animation List
+			
 			if (o->GetAnimationObject()) {
 				AnimationObject* animObj = o->GetAnimationObject();
 				mAnimationList.emplace_back(animObj);
 				//TODO may it is not a good position to run
-				UpdateCurrentFrames(dt);
-
+				
+				int currentFrame = animObj->GetCurrentFrame();
 				mMesh = o->GetRenderObject()->GetMesh();
 				mAnim = animObj->GetAnimation();
 				mShader = o->GetRenderObject()->GetShader();
-				int currentFrame = animObj->GetCurrentFrame();
-
-				const Matrix4* bindPose = mMesh->GetBindPose().data();
+				
 				const Matrix4* invBindPose = mMesh->GetInverseBindPose().data();
 				const Matrix4* frameData = mAnim->GetJointData(currentFrame);
-				vector<Matrix4> frameMatrices;
-
 				
-				for (unsigned int a = 0; a < mMesh->GetJointCount(); ++a) {
-					frameMatrices.emplace_back(frameData[a] * invBindPose[a]);
-				}
+				const int* bindPoseIndices = mMesh->GetBindPoseIndices();
+				std::vector<std::vector<Matrix4>> frameMatricesVec;
+				for (unsigned int i = 0; i < mMesh->GetSubMeshCount(); ++i) {
 
+					
+					Mesh::SubMeshPoses pose;
+					mMesh->GetBindPoseState(i, pose);
+					
+
+					vector<Matrix4> frameMatrices;
+					for (unsigned int i = 0; i < pose.count; ++i) {
+						/*
+						We can now grab the correct matrix for a given pose.
+						Each matrix is relative to a given joint on the original mesh.
+						We can perform the lookup for this by grabbing a set of indices
+						from the mesh.
+						*/
+						int jointID = bindPoseIndices[pose.start + i];
+
+						Matrix4 mat = frameData[jointID] * invBindPose[pose.start + i];
+
+						frameMatrices.emplace_back(mat);
+					}
+					frameMatricesVec.emplace_back(frameMatrices);
+				}
 				o->GetRenderObject()->SetAnimation(o->GetAnimationObject()->GetAnimation());
 				o->GetRenderObject()->SetMaterial(o->GetAnimationObject()->GetMaterial());
-				o->GetRenderObject()->SetCurrentFrame(o->GetAnimationObject()->GetCurrentFrame());
-				o->GetRenderObject()->SetFrameMatrices(frameMatrices);
+				o->GetRenderObject()->SetCurrentFrame(currentFrame);
+				o->GetRenderObject()->SetFrameMatricesVec(frameMatricesVec);
+				
+				frameMatricesVec.clear();
+				
+				
+				
+
+				if (o->GetName() == "Guard") {
+					mGuardList.emplace_back((GuardObject*)o);
+					/*std::cout << "find the guard" << std::endl;*/
```

Comment:
Best to delete commented out code. If you still need it for some reason, mark it with your name, so we know who to chase if it never gets done ;)

Reply:
Got it!